src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -479,948 +575,1525 @@ public void LastIndexOf()
+                Assert.Throws<ArgumentNullException>("items", () => ((IImmutableList<int>)array).RemoveRange(items: null));
+                Assert.Throws<ArgumentNullException>("items", () => ((IImmutableList<int>)array).RemoveRange(items: null, equalityComparer: comparer));
+
+                // Enumerable overloads, lhs is default and rhs is null
+                Assert.Throws<NullReferenceException>(() => s_emptyDefault.RemoveRange(items: null));
+                Assert.Throws<NullReferenceException>(() => s_emptyDefault.RemoveRange(items: null, equalityComparer: comparer));
+                Assert.Throws<InvalidOperationException>(() => ((IImmutableList<int>)s_emptyDefault).RemoveRange(items: null));
+                Assert.Throws<InvalidOperationException>(() => ((IImmutableList<int>)s_emptyDefault).RemoveRange(items: null, equalityComparer: comparer));
+            });
+        }
+
+        [Fact]
+        public void RemoveRangeEnumerableRegression()
+        {
+            // Validates that a fixed bug in the inappropriate adding of the Empty
+            // singleton enumerator to the reusable instances bag does not regress.
+            
+            IEnumerable<int> oneElementBoxed = s_oneElement;
+            IEnumerable<int> emptyBoxed = s_empty;
+            IEnumerable<int> emptyDefaultBoxed = s_emptyDefault;
+
+            Assert.Throws<NullReferenceException>(() => s_emptyDefault.RemoveRange(emptyBoxed));
+            Assert.Throws<NullReferenceException>(() => s_emptyDefault.RemoveRange(emptyDefaultBoxed));
+            Assert.Throws<InvalidOperationException>(() => s_empty.RemoveRange(emptyDefaultBoxed));
+
+            Assert.Equal(oneElementBoxed, oneElementBoxed);
+        }
+
+        [Theory]
+        [MemberData(nameof(RemoveAllData))]
+        public void RemoveAll(IEnumerable<int> source, Predicate<int> match)
+        {
+            var array = source.ToImmutableArray();
+            var expected = source.Where(i => !match(i));
+            Assert.Equal(expected, array.RemoveAll(match));
+        }
+
+        public static IEnumerable<object[]> RemoveAllData()
+        {
+            yield return new object[] { Enumerable.Range(1, 10), new Predicate<int>(i => i % 2 == 0) };
+            yield return new object[] { Enumerable.Range(1, 10), new Predicate<int>(i => i % 2 == 1) };
+            yield return new object[] { Enumerable.Range(1, 10), new Predicate<int>(i => true) };
+            yield return new object[] { Enumerable.Range(1, 10), new Predicate<int>(i => false) };
+            yield return new object[] { s_empty, new Predicate<int>(i => false) };
+        }
+
+        [Theory]
+        [MemberData(nameof(Int32EnumerableData))]
+        public void RemoveAllInvalid(IEnumerable<int> source)
+        {
+            var array = source.ToImmutableArray();
+
+            Assert.Throws<ArgumentNullException>("match", () => array.RemoveAll(match: null));
         }
 
   
```

Comment:
Same here, lets look into bringing this into buildtools so we don't have to work around its absence.